lib/notice.js

```diff
@@ -0,0 +1,57 @@
+const { MessageEmbed } = require('discord.js')
+const config = require('./config')
+
+const userCooldowns = new Map()
+const channelCooldowns = new Map(config.channels.notices.map(v => [v, 0]))
+
+function cleanup () {
+  const now = Date.now()
+  for (const [user, timeout] of userCooldowns) {
+    if (timeout < now) {
+      userCooldowns.delete(user)
+    }
+  }
+}
+
+function isEnabled () {
+  return this.enabled && (this.embed.title || this.embed.description)
+}
+
+function processMessage (msg) {
+  if (!this.isEnabled()) {
+    return
+  }
+
+  if (msg.member.roles.highest.name !== '@everyone') {
+    // assume anyone with a rank can use their eyes
+    return
+  }
+
+  const now = Date.now()
+  if (userCooldowns.get(msg.author.id) > now) {
+    userCooldowns.set(msg.author.id, now + this.userCooldown)
+    return
+  }
+
+  const channelCooldown = channelCooldowns.get(msg.channel.name)
+  if (channelCooldown === undefined || channelCooldown > now) {
+    return
+  }
+
+  channelCooldowns.set(msg.channel.name, now + this.channelCooldown)
+  userCooldowns.set(msg.author.id, now + this.userCooldown)
+
+  msg.channel.send(this.embed)
+}
+
+module.exports = {
+  processMessage,
+  isEnabled,
+  cleanup,
+
+  enabled: false,
+  channelCooldown: 15_000,
+  userCooldown: 120_000,
+
+  embed: new MessageEmbed().setColor(0xec644b) // why isn't this brand orange?
```

Comment:
This comment isnt really necessary. Also you should use the buildEmbed() method from common, the point of it is so embeds are consistent (and that we can change the design of the embed from one place).

Reply:
I explicitly didn't use createEmbed because I would have to unset half the defaults it is setting, and if we change it in the future we would have to change it back here.

Reply:
You only need to change the thumbnail, and I wouldn't mind removing it from the main function, I do not like it anyway.

Reply:
I would need to do
```javascript
createEmbed()
    .setThumbnail(undefined)
    .setFooter(undefined)
```
plus if we ever added a new default we would need to change it here too

Reply:
You dont need to remove footer tho. Imo keeping the bot footer is fine. The footer command will override it anyway (even though, on that note i dont think we need to override it and I would rather just remove the footer command). And for thumbnail, as I said I would be fine with removing it from the main method completely. And for other defaults, I dont think there will be any.

Reply:
I really don't want the footer by default since it about doubles the vertical space the embed consumes and is completely useless. I would be in favor of removing it from the default embed too.

The footer command is so we can abuse the timestamp to get a sentence ending in the user's local timezone, which is very useful during downtime when we have an ETA.

Reply:
Sure, that sounds fine as well. I don't get that part about footer and timestamp tho. Its just so the users see that *something* is happening? Even when they dont understand the message?

Reply:
When you have footer & timestamp both set it renders as `<footer> • <timestamp in user's local timezone>`, so in the downtime from a few days ago I had it set as `They expect the game to come back online • Today at 7:00 AM`